extensions/adobe/experience/mobile-lifecycle-details.schema.json

```diff
@@ -0,0 +1,279 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/mixins/mobile-lifecycle-details",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Mobile Lifecycle Details",
+  "type": "object",
+  "description": "Application lifecycle information recorded via an ExperienceEvent that is specific to the Mobile channel. Includes field definitions for automatically collected data by the AEP Edge mobile extension library.",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "definitions": {
+    "mobile-lifecycle-details": {
+      "properties": {
+        "xdm:mobileApplication": {
+          "title": "Mobile Application",
+          "$ref": "https://ns.adobe.com/xdm/datatypes/mobile-application",
+          "description": "Application information related to an ExperienceEvent, including the name of the application, app version, launches, and closures."
+        },
+        "xdm:device": {
+          "title": "Device",
+          "$ref": "https://ns.adobe.com/xdm/context/device",
+          "description": "Information about the mobile device the application lifecycle event observation occurred in, including device type and model."
+        }
+      }
+    },
+    "environment-type": {
+      "properties": {
+        "xdm:environment": {
+          "type": "object",
+          "properties": {
+            "xdm:type": {
+              "title": "Type",
+              "$ref": "https://ns.adobe.com/xdm/context/environment#/definitions/type",
+              "description": "The type of the application environment."
+            }
+          }
+        }
+      }
+    },
+    "environment-operatingSystem": {
+      "properties": {
+        "xdm:environment": {
+          "type": "object",
+          "properties": {
+            "xdm:operatingSystem": {
```

Comment:
@kevinlind , $ref is used to refer a schema or a definition from the local or an external schema. In this case you are cloning the properties from the "environment" data type schema and they need to be cloned as such (the way they are defined in the data type). you are using reference as "https://ns.adobe.com/xdm/context/environment#/definitions/operatingSystem"
which is wrong as "type" is not defined as a definition in the environment schema. Its just a scalar property. So, in order to define that here you would have use the same property here as well like below:
    "environment-operatingSystem": {
      "properties": {
        "xdm:environment": {
          "type": "object",
          "properties": {
             "xdm:operatingSystem": {
             "title": "Operating system",
             "type": "string",
             "description": "The name of the operating system used when the observation was made. The attribute should 
               not contain any version information such as '10.5.3', but instead contain 'edition' designations such as 
               'Ultimate' or 'Professional'."
               },
The same comment applies to all the properties here

Reply:
Thanks for the clarification. My concern with cloning the scalar in the mixin is if the property is changed in the Environment schema, such as a new value added to the connectionType enum, then the cloned property in the mixin won't automatically update. It sounds like that is unavoidable unless the Environment schema is changed to make each property a separate definition (or at least the properties which can change such as the enum types.).

I'll update the properties for this mixin as you suggested. However, is it possible to change the Environment schema to define some or all of the properties as definitions so they may be referenced with $ref, or is that considered a breaking change?

Reply:
@kevinlind , your concern is a very valid concern and we are aware of it. We have plans to make these properties as definitions in future but before that, we have to come up with a best strategy of doing it with minimal impact on the schemas.

Reply:
That's great to hear. I'm wondering if I should change my definition of the Mobile Application schema to declare each property as its own definition? I don't have any use for that now, but it would be nice to have the flexibility for any future need.